clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -28,18 +28,28 @@
     private final String host;
     private final int port;
     private final String rack;
+    private final boolean hasPriority;
```

Comment:
This is an interesting approach. It makes sense for the consumer case where we have a single connection to the coordinator, but I wonder if it would be more flexible to do this on a per-request basis? One thing I was thinking about is whether we'd ever need to prioritize metadata requests. Perhaps fetch requests are the only request types we should ever require allocation in the memory pool?